lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -71,6 +72,16 @@ function initializeImportMeta(meta, context, loader) {
 
   meta.url = url;
 
+  ObjectDefineProperty(meta, 'require', {
+    __proto__: null,
+    enumerable: true,
+    configurable: true,
+    get() {
+      const { Module: { createRequire } } = require('internal/modules/cjs/loader');
+      return createRequire(url);
+    },
+  });
```

Comment:
I think we don't want to re-create the require on every access, so instead overwrite the prop after the first call with the value created, like https://github.com/nodejs/node/blob/main/lib/internal/modules/package_json_reader.js#L73-L77